lib/perl/Genome/InstrumentData/AlignmentResult/Command/RecreatePerLaneBam.pm

```diff
@@ -136,7 +139,7 @@ sub _extract_readgroup_bam {
 
 
 sub _compare_flagstat {
-    my ($self, $temp_bam, $flagstat) = @_;
+    my ($self, $temp_bam, $flagstat, $diff) = @_;
```

Comment:
I think that this whole "diff" thing should just be a separate function.  However, if you keep it like this then I think calling `$diff` something else, maybe `$ignore_duplicates`, would be a better description.